streams/src/main/java/org/apache/kafka/streams/state/internals/CachingWindowStore.java

```diff
@@ -85,8 +86,13 @@ public void init(final StateStoreContext context, final StateStore root) {
     }
 
     private void initInternal(final InternalProcessorContext<?, ?> context) {
+        final String prefix = StreamsConfig.InternalConfig.getString(
```

Comment:
Could we use `ProcessorContextUtils#changelogFor` here as well? Ditto below

Reply:
It could either be a `StateStoreContext` of a `ProcessorContext` and there are different methods for each. I am not sure why as they seem to do similar things. but I didn't want to make reaching changes here. I can just file a ticket to consolidate those methods into one maybe?

Reply:
In the future we would remove the deprecated `init(final ProcessorContext context, final StateStore root)` and then we only need `ProcessorContextUtils#changelogFor(StateStoreContext..)`.

Sounds good for filing a JIRA for consolidating to the `changelogFor`, we can do that when we remove the deprecated `init` function.

Reply:
https://issues.apache.org/jira/browse/KAFKA-13588